src/main/java/org/jabref/gui/search/SearchTextField.java

```diff
@@ -13,7 +15,21 @@
     public static TextField create() {
         CustomTextField textField = (CustomTextField) TextFields.createClearableTextField();
         textField.setPromptText(Localization.lang("Search") + "...");
-        textField.setLeft(IconTheme.JabRefIcon.SEARCH.getGraphicNode());
+        Node node = IconTheme.JabRefIcon.SEARCH.getGraphicNode();
+        node.setStyle("-fx-text-fill: #00ffff");
```

Comment:
-fx-fill should work, but do you really think that changing the icon color is a good idea? How intuitive (and important) is it to signal an advanced search mode by magenta?

Reply:
Unfortunately, `fx-fill` doesn't help. There seems to be a deeper problem with exactly this icon.

Actually, I am quite indifferent to this, but @koppor really wants it.

Reply:
I really like the proposal at https://github.com/JabRef/jabref/issues/3535#issuecomment-352068826. However, JabRef had this functionality for at least one year. Animation at https://github.com/JabRef/jabref/issues/3535#issuecomment-363813333. - Simon and I had a long discussion on that. The decision was that we should present to the user that there are different search modes. They should "just work", but for the pros, there should be a hint which mode is active. Not that someone wonders why the search does not work. -- Furthermore, that icon makes it less hard to debug the search. It is clear, whether the search term is a grammar-based search or just plain text string search.

Reply:
ok, I see your point. It is just again a feature that needs documentation before anybody understands what's going it. 

Butt if `fx-fill` and `fx-text-fill` don't work, I'm running out of ideas. I usually use a combination out of both and it worked without any problems so far. Maybe it works if you base this PR on the maintable-beta branch. I changed a few things regarding javafx icons there.

Reply:
I have tried some more (not the maintable branch, though) with no success. I think this is down to `CustomTextField` from controls-fx. It offers the functionality to add a node at the left or right side, but it seems to be impossible to change the style of this node.

Because of this, I've gone back to the previous version: A Swing icon next to the text field and everything just works. I know this is a step back, but if the `CustomTextField` cannot be made to use our icon theme, then we should probably migrate the whole `GlobalSearchBar` to JavaFX instead. Until this happens, I think it is wiser to use icons that preserve our color theme.

@tobiasdiez @koppor: I'll now leave it open to you to continue arguing, merge, or close this PR. I am fine with any solution.